fieldtracing/fieldtracing.cpp

```diff
@@ -996,114 +1010,202 @@ namespace FieldTracing {
                      ) {
                         cellNeedsContinuedBWTracing[n] = 0;
                         cellBWTracingCoordinates[n] = {0,0,0};
-                        cellBWConnection[n] = TracingLineEndType::OPEN;
+                        cellBWConnection[n] += TracingLineEndType::OPEN;
+                        break;
+                     }
+                     
+                     // If we exceed the max tracing distance we're probably looping
+                     if(cellBWRunningDistance[n] > maxTracingDistance) {
+                        cellNeedsContinuedBWTracing[n] = 0;
+                        cellBWTracingCoordinates[n] = {0,0,0};
+                        cellBWConnection[n] += TracingLineEndType::LOOP;
+                        #pragma omp critical
+                        {
+                           warnMaxDistanceExceeded = true;
+                        }
                         break;
                      }
                      
+                     // If we are still in the race for flux rope...
+                     if(cellBWConnection[n] < TracingLineEndType::N_TYPES) {
+                        creal distance = sqrt(
```

Comment:
So, as a reminder form our zoom: Distance can be stored in an array to be placed in the fluxrope cellparam, and you only need to do one MPI_ALLREDUCE at the end with MPI_MAX in order to get the maximum followed distance from all mpi-tasks for each cell.

Reply:
Yep, thanks! So do we store the distance in m or in curvature radii? And the max number of curvature radii traced as a VLSV parameter?